### PR TITLE
[MM-52943] fix: enable Elasticsearch properly on restart

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

During some investigation of Elasticsearch issues, I found out that the Elasticsearch engine wasn't properly booting on a server restart. Putting a breakpoint on the `Start()` method of the engine showed that this was only called on a configuration or license change (as it should be), but the initial call wasn't going through because the `IsActive()` checks for the engine to be enabled, and the `Start()` method takes care of that (but it's never called). 

I've added a new method to the interface `IsEnabled()` that will return if the engine is enabled regardless if it is currently running or not.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-52943

#### Screenshots
--

#### Release Note

```release-note
fix: properly start elasticsearch on startup if enabled
```

### CODE REVIEW

File: server/platform/services/searchengine/bleveengine/bleve.go [hunk 1]
```diff
@@ -219,6 +219,10 @@ func (b *BleveEngine) Stop() *model.AppError {
 	return b.closeIndexes()
 }
 
+func (b *BleveEngine) IsEnabled() bool {
+	return b.IsActive()
```
Comment (agnivade) [inline server/platform/services/searchengine/bleveengine/bleve.go:223]: This should be return `b.IsIndexingEnabled()`

**Review by agnivade — APPROVED** [review]

**Comment by isacikgoz** [comment]
@toninis Seems like `Enterprise CI/tests` is pulling `master` branch of EE code, can you take a look?

**Comment by fmartingr** [comment]
> @toninis Seems like `Enterprise CI/tests` is pulling `master` branch of EE code, can you take a look?

Enterprise tests would probably fail until https://github.com/mattermost/enterprise/pull/1489 is merged, right? What I'm not sure is why the mysql tests are failing.

**Review by isacikgoz — APPROVED** [review]
I think it's just the same falky test.

**Comment by amyblais** [comment]
@fmartingr Will this need QA review?

**Comment by fmartingr** [comment]
> @fmartingr Will this need QA review?

I've tested it properly since it's a small bug, so I don't think that's necessary.

**Comment by amyblais** [comment]
@fmartingr This looks good to merge.

**Comment by amyblais** [comment]
> Sorry, I'm not familiar with the mattermost workflows and I though someone else would merge it.

Ah ok, normally the PR submitter own merging the PR 👍

**Pull request merged**